modules/tutorials/pages/java-tutorial/adding-enrollments.adoc

```diff
@@ -0,0 +1,202 @@
+
+        // Retrieve the records
+        JsonObject hilary = retrieveStudent(cluster,"Hilary Smith");    // <5>
+        JsonObject graphic_design = retrieveCourse(cluster, "graphic design");    // <5>
+        JsonObject art_history = retrieveCourse(cluster, "art history");    // <5>
+
+        String currentDate = LocalDate.now().format(DateTimeFormatter.ISO_DATE);     // <6>
+
+        // Create Hilary's enrollments
+
+        JsonArray enrollments = JsonArray.create();     // <7>
+
+        enrollments.add(JsonObject.create().put("course-id", graphic_design.getString("id"))
+                .put("date-enrolled", currentDate));    // <8>
+
+        enrollments.add(JsonObject.create().put("course-id", art_history.getString("id"))
+                .put("date-enrolled", currentDate));    // <8>
+
+        hilary.put("enrollments", enrollments);     // <9>
+
+        student_records.upsert(hilary.getString("id"), hilary);    // <10>
+
+        cluster.disconnect();
+
+    }
+
+    private static JsonObject retrieveStudent(Cluster cluster, String name) throws CouchbaseException {
+
+        final QueryResult result = cluster.query("select META().id, src.* " +
+                        "from `student-bucket`.`art-school-scope`.`student-record-collection` src " +
+                        "where src.`name` = $name",
+                QueryOptions.queryOptions()
+                        .parameters(JsonObject.create().put("name", name)));
+
+        return result.rowsAsObject().get(0);
+
+    }
+
+    private static JsonObject retrieveCourse(Cluster cluster, String course) throws CouchbaseException {
+
+        final QueryResult result = cluster.query("select META().id, crc.* " +
+                "from `student-bucket`.`art-school-scope`.`course-record-collection` crc " +
+                "where crc.`course-name` = $courseName",
+                QueryOptions.queryOptions()
+                        .parameters(JsonObject.create().put("courseName", course)));
+
+        return result.rowsAsObject().get(0);
+
+    }
+
+}
+
+----
+<1> As you'll remember from our xref:java-tutorial/first-couchbase-application.adoc#connecting-to-the-database[first example], the application first has to connect to the Couchbase cluster
+
+<2> Then it picks up the correct `Bucket` from the cluster.
+
+<3> Do you remember why we need this `waitUntilReady` function is needed when connecting to the server?
++
+{empty}
++
+[%collapsible]
+.Answer
+====
+{why-waitUntilReady}
+====
++
+{empty}
+<4> Here again, we pick up the student collection from the `art-school-scope`; you'll need it later to change Hilary's record and write it back to the collection.
+
+<5> There are three method calls here:
++
+--
+. Retrieve Hilary's student record
+. Retrieve the `graphic design` course record
+. Retrieve the `art history` course record
+--
++
+Each method uses a `N1QL` call to retrieve a single record from its respective collection.
```

Comment:
I don't think we need to format N1QL as code.

Reply:
fixed